### PR TITLE
`mockLiveUpdate`

### PR DESCRIPTION
**Pull request by oliverlloyd**
## What does this change?
This adds a function `mockLiveUpdate` on the `window` object. This function is a reference to the `onSuccess` function that gets called after a live update request is made on a live blog

## Why?
We need a way to test liveblogs using Cypress. By exposing this code like this it becomes available to our test scripts without them needing to depend on polling.

## Why `window`?
Using the `window` object is convenient because it is accessible to Cypress using [cy.window()](https://docs.cypress.io/api/commands/window)

### CODE REVIEW

File: dotcom-rendering/window.guardian.d.ts [hunk 1]
```diff
@@ -68,6 +68,11 @@ declare global {
 			onConsentChange?: (fn: Callback) => void;
 			getConsentFor?: (fn: Callback) => void;
 		};
+		mockLiveUpdate: (data: {
+			numNewBlocks: number;
+			html: string;
+			mostRecentBlockId: string;
+		}) => void;
```
Comment (mxdvl) [inline dotcom-rendering/window.guardian.d.ts:75]: If you import onSuccess from that file, you can keep everything in sync.

```suggestion
		mockLiveUpdate: typeof onSuccess
```

**Review by mxdvl — APPROVED** [review]
I think it’s best to have our type declaration get its type from functions defined in our code, but otherwise I think it’s good to be able to mock!

File: dotcom-rendering/window.guardian.d.ts (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/window.guardian.d.ts]: Sorry, you’ll also need to import the method in this file!

File: dotcom-rendering/window.guardian.d.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ declare global {
 			onConsentChange?: (fn: Callback) => void;
 			getConsentFor?: (fn: Callback) => void;
 		};
+		mockLiveUpdate: typeof onSuccess
```
Comment (jamesgorrie) [inline dotcom-rendering/window.guardian.d.ts:71]: Does this need to be imported here?

File: dotcom-rendering/window.guardian.d.ts (hunk 1 above)
Comment (oliverlloyd) [reply dotcom-rendering/window.guardian.d.ts]: I created `LiveUpdateType` here instead. Like this we don't need to export the function, which feels like a win in terms of reducing complexity in the code

File: dotcom-rendering/window.guardian.d.ts (hunk 2 above)
Comment (oliverlloyd) [reply dotcom-rendering/window.guardian.d.ts:71]: As above, I created `LiveUpdateType` instead

**Pull request merged**